orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -478,6 +480,9 @@ def parse_outputs_for_onnx_export_and_extract_schema(module, inputs, kwargs):
         output_names, output_dynamic_axes = _parse_outputs_and_extract_names_and_dynamic_axes(sample_outputs)
     if is_train_mode:
         module.train()
-
+    output_schema = _extract_schema(sample_outputs)
+    if is_deepcopy:
+        del model_copy
+        gc.collect()
```

Comment:
I'm just a little concerned that this is needed explicitly to be called here. I am guessing that this really helps in freeing up memory.

Reply:
Yes. Without gc.collect() the memory doesn't get freed up since it is a big python object.